include/xtensor/xfiles_array.hpp

```diff
@@ -0,0 +1,233 @@
+#ifndef XTENSOR_FILES_ARRAY_HPP
+#define XTENSOR_FILES_ARRAY_HPP
+
+#include <istream>
+#include <fstream>
+#include <iostream>
+#include <vector>
+#include <array>
+
+#include "xarray.hpp"
+#include "xcsv.hpp"
+#include "xio.hpp"
+
+namespace xt
+{
+    template <class S>
+    void reopen(S& stream, const std::string& path)
+    {
+        if (stream.is_open())
+        {
+            stream.close();
+            stream.clear();
+        }
+        stream.open(path);
+    }
+
+    template <class T1, class T2, class S1, class S2>
+    void remap(T1** file_array, T2& array, S1& in_stream, S2& out_stream)
+    {
+        if (*file_array != NULL)
+        {
+            delete *file_array;
+            *file_array = NULL;
+        }
+        *file_array = new T1(array, in_stream, out_stream);
+    }
+
+    template <class EC>
+    class xfiles_array;
+
+    template <class EC>
+    struct xcontainer_inner_types<xfiles_array<EC>>
+    {
+        using const_reference = const EC&;
+        using reference = EC&;
+        using size_type = std::size_t;
+        using storage_type = EC;
+        using temporary_type = xfiles_array<EC>;
+    };
+
+    template <class EC>
+    struct xiterable_inner_types<xfiles_array<EC>>
+    {
+        using inner_shape_type = std::vector<size_t>;
+        using const_stepper = xindexed_stepper<xfiles_array<EC>, true>;
+        using stepper = xindexed_stepper<xfiles_array<EC>, false>;
+    };
+
+    template <class EC>
+    class xfiles_array: public xaccessible<xfiles_array<EC>>,
+                        public xiterable<xfiles_array<EC>>
+    {
+    public:
+
+        using const_reference = const EC&;
+        using reference = EC&;
+        using self_type = xfiles_array<EC>;
+        using iterable_base = xconst_iterable<self_type>;
+        using const_stepper = typename iterable_base::const_stepper;
+        using stepper = typename iterable_base::stepper;
+        using inner_types = xcontainer_inner_types<self_type>;
+        using size_type = typename inner_types::size_type;
+        using storage_type = typename inner_types::storage_type;
+        using value_type = storage_type;
+        using pointer = value_type*;
+        using const_pointer = const value_type*;
+        using difference_type = std::ptrdiff_t;
+        using shape_type = std::vector<size_t>;
+
+        template <class O>
+        const_stepper stepper_begin(const O& shape) const noexcept;
+        template <class O>
+        const_stepper stepper_end(const O& shape, layout_type) const noexcept;
+
+        template <class O>
+        stepper stepper_begin(const O& shape) noexcept;
+        template <class O>
+        stepper stepper_end(const O& shape, layout_type) noexcept;
+
+        const shape_type& shape() const
+        {
+            return m_shape;
+        }
+
+        xfiles_array(): m_file_array(NULL)
```

Comment:
Why is `m_file_array` a pointer instead of a plain object? If you keep the pointer, you need to carefully implement the copy and move semantics, otherwise you'll have a lot of crashes when using the xfiles_array:

xfiles_array<xdisk_array<double>>> a1, a2;
// ....
a1 = a2; 
// => a1.m_file_array is never destroyed before beeing assigned a2.m_array.
// => a2.m_array will be destroyed twice.

Reply:
Thanks, it simplifies a lot indeed!